src/credentials/FabricTable.h

```diff
@@ -174,6 +175,19 @@ class DLL_EXPORT FabricInfo
         return Credentials::ExtractPublicKeyFromChipCert(mRootCert, publicKey);
     }
 
+    /**
+     * Get the effective NotBefore time for a certificate chain, which is
+     * the latest NotBefore time in the chain.  Note that chip TLV certificate
+     * format stores the  X.509/RFC5280 special time 99991231235959Z 'no
+     * well-defined expiration date' as 0.  In this context, we can interpret
+     * that as either NotBefore == CHIP epoch or NotBefore == Don't care, and
+     * this will evaluate the same in both cases.
+     *
+     * @param certChainNotBefore (out) effective NotBefore time for the certiticate chain in seconds from CHIP epoch
+     * @return CHIP_NO_ERROR on success, else an appropriate CHIP_ERROR
+     */
+    CHIP_ERROR GetNotBeforeChipEpochTime(chip::System::Clock::Seconds32 & certChainNotBefore) const;
```

Comment:
This should be a method in CHIPCertificateSet, since we are actually working to move away from "keeping a pending fabric info during fail-safe" to move to FabricTable owning the temporary state... An alternative is to have the method in FabricTable, taking a FabricIndex